compact_str/src/utility.rs

```diff
@@ -0,0 +1,58 @@
+use core::fmt;
+use std::io::{Cursor, Result, Write};
+
+/// A special kind of sink that records the size of bytes writen into it.
+#[derive(Debug, Default)]
+pub struct Sink(pub usize);
+impl Sink {
+    pub fn count(args: fmt::Arguments) -> usize {
+        let mut sink = Sink::default();
+        write!(&mut sink, "{}", args).unwrap();
+        sink.0
+    }
+}
```

Comment:
This is pretty neat! See above though, I don't think we need to do this since `CompactStr` impls `fmt::Write`